src/client/AM.PA.MacOs/PersonalAnalytics/Shared/DataObjectController.swift

```diff
@@ -74,236 +83,124 @@ class DatabaseController{
 **/
 class DataObjectController: NSObject{
```

Comment:
I think this whole class is kind of bad practice. Probably each tracker should have its own data controller class which is responsible for saving its data to the database. This way when we add more trackers we dont have to keep adding functions in this class.

Also let's move DatabaseController to it's own file.

Reply:
I agree. In the Windows version, we have a Queries class for each Tracker which stores all the queries that access the database.

Reply:
Totally agree. I already prepared this in the [mac-tracker-refactoring](https://github.com/sealuzh/PersonalAnalytics/tree/mac-tracker-refactoring) branch. I created a separate Queries class for each tracker and moved the methods in there.